src/graph/create-entity.ts

```diff
@@ -49,8 +50,8 @@ type CreateEntityParams = DefaultProperties & {
  * @returns – {@link CreateResult}
  * @throws Will throw an error if any provided ID is invalid
  */
-export const createEntity = ({ name, description, cover, properties, types }: CreateEntityParams): CreateResult => {
-  const id = generate();
+export const createEntity = ({ id: providedId, name, description, cover, properties, types }: CreateEntityParams): CreateResult => {
+  const id = providedId ?? generate();
```

Comment:
Should we validate that `providedId` is a valid id?

Reply:
Yes! added